### DISCUSSION TITLE
Docker install on QNAP NAS - where to put the /config/config.yml file

### DISCUSSION THREAD

**Discussion by cralwalker** [blakeblackshear/frigate#2563] [Ask A Question]
When I install v0.1.2 (or v0.1.1) through QNAPs Container Station app, even though I have placed a configuration in /config/config.yml, I get the following response:

Traceback (most recent call last):
  File "detect_objects.py", line 12, in <module>
    with open('/config/config.yml') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/config/config.yml'

I have placed the same directory&file in various places, root, user root, in the PATHs specified at installation "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"

and the file is still not picked up. My apologies if I am doing it wrong?

**Comment by blakeblackshear** [comment]
Where are these version numbers coming from? I don't recognize them. I don't know anything about Qnap container station. Can you just run a docker command instead?

**Comment by cralwalker** [comment]
Fair enough, thanks for your response.

Being a noob script bunny, I used the docker command from Docker Hub (https://hub.docker.com/r/blakeblackshear/frigate/#!)
I got the following response

[~] # docker pull blakeblackshear/frigate
Using default tag: latest
Error response from daemon: manifest for blakeblackshear/frigate:latest not found: manifest unknown: manifest unknown

Again, apologies if I am doing it wrong?

**Reply by blakeblackshear** [reply]
There is no latest tag. Please look at the installation docs for frigate.

**Reply by smartthings650** [reply]
I am having the same issue using openmediavault I have tried almost everywhere still the same error FileNotFoundError: [Errno 2] No such file or directory: '/config/config.yml'  hope someone can help with this

**Comment by uklatt** [comment]
Hi,
you need to add a directory mapping in container station when creating the container!
You can do this only when creating the container.
Then you can add the config file in this directory.
You must also map the TCP port 5000 for Webaccess.
Uwe

**Comment by alienatedsec** [comment]
It seems to be an ongoing issue for some - I managed to install Frigate container on my QNAP (x86), and wanted to share my experience. I ran it by executing the following from the console.

```
docker run -d --name=frigate_13beta2 --shm-size=256m -e LIBVA_DRIVER_NAME=i965 --device=/dev/apex_0:/dev/apex_0 --device=/dev/dri/renderD128 --restart=unless-stopped --env=TZ=Europe/London --volume=/share/Container/frigate/config:/config:rw --volume=/share/frigate/media:/media/frigate:rw --network=bridge --privileged --workdir=/opt/frigate -p 1935:1935 -p 5000:5000 -p 8554:8554 -p 8555:8555 -p 8555:8555/udp --label='com.qnap.qcs.network.mode=nat' --label='com.qnap.qcs.gpu=False' --memory="4g" --cpus="2" --detach=true -t ghcr.io/blakeblackshear/frigate:0.13.0-beta2
```

In the above scenario, it will be a NATted address, so Frigate will be available at `http://[QNAP IP]:5000`

![image](https://github.com/blakeblackshear/frigate/assets/73167064/03fbb74a-98a3-4c0e-9cb3-de2e018704a9)

While the above is for the latest beta, it also works with the `ghcr.io/blakeblackshear/frigate:stable` flag
- `--name=frigate_13beta2` my preference
- `-e LIBVA_DRIVER_NAME=i965` recommended GPU driver for my processor `J4125`
- `--device=/dev/apex_0:/dev/apex_0` is my mini PCIe Coral; however, the `--privileged` flag will likely allow us to use it anyway. 

If you need more info on how to install [https://coral.ai/products/m2-accelerator-ae](https://coral.ai/products/m2-accelerator-ae) on your QNAP then let me know.

- `--device=/dev/dri/renderD128` intel GPU; however, the `--privileged` flag will likely allow us to use it anyway.
- `--volume=/share/Container/frigate/config:/config:rw` is the location of my config files - you need to create the folder in that location `/share/Container/frigate/config`
- `--volume=/share/frigate/media:/media/frigate:rw` is the location of the recordings - you need to create the folder in that location `/share/frigate/media`

Four cameras, and planning more :) Thanks to @blakeblackshear and the team. Worth updating or referencing in documentation.

![image](https://github.com/blakeblackshear/frigate/assets/73167064/78989562-b231-46c6-a7fd-65a5e58a46ef)

**Reply by Wildcat1** [reply]
How do you setup shared folders for the container in Container Station 3.0.9 using the GUI ? I see no option (unless there is an Environment Variable to add this functionality) to add the shared folders at point of Container create. Thanks
